pkg/tecdsa/tecdsa.go

```diff
@@ -0,0 +1,19 @@
+package tecdsa
+
+import "time"
+
+const (
+	DefaultPreParamsPoolSize              = 50
+	DefaultPreParamsGenerationTimeout     = 2 * time.Minute
+	DefaultPreParamsGenerationConcurrency = 1
+)
+
+// Config carries the config for threshold ECDSA protocol.
+type Config struct {
```

Comment:
Are we sure this type should be declared here? Pre-params pool and its config are something DKG-specific and declaring that type here makes `pkg/tecdsa` package silently dependent on `pkg/tecdsa/dkg` while the proper direction is something opposite (`pkg/tecdsa/dkg` depends on `pkg/tecdsa` by using the `PrivateKeyShare` type). I see that `dkg.NewExecutor` now exposes those params directly so there is probably no point to move `Config` there. Looks like we need it just for convenience when we are parsing config flags/file so maybe we should put `Config` directly into `pkg/tbtc` and expect it as a parameter of `tbtc.Initialize`?

Reply:
I placed the config in `pkg/tecdsa` to not create a dependency from `pkg/tecdsa` to `pkg/tecdsa/dkg`. If we had this config in `pkg/tecdsa/dkg`, then we'd need to accept it as a parameter of `tbtc.Initialize` and we'd need to add an import from `pkg/tecdsa` to `pkg/tecdsa/dkg`.

Putting aside the direction of dependencies, from the protocol perspective, this structure fits best in `pkg/tecdsa/dkg` because it contains only DKG-specific parameters. `pkg/tecdsa` is not that bad because we are still in `tecdsa` land, we may add more parameters to this config, and we'll have to make this code interact with the rest of the protocols executed by the client to pause pre-params generation (separate set of PRs) so `pkg/tecdsa/dkg` is too deep.

I can put this struct in `tbtc` but then `dkg.NewExecutor` needs to accept params directly (not a struct). To me both options are fine. The way I look at it is that these are `tecdsa` parameters so we are not creating a silent dependency from `pkg/tecdsa` to `pkg/tecdsa/dkg`. We are rather having `pkg/tecdsa/dkg` accepting general tECDSA parameters. But I am happy to change it. What do you think?

Reply:
>  The way I look at it is that these are tecdsa parameters so we are not creating a silent dependency from pkg/tecdsa to pkg/tecdsa/dkg. We are rather having pkg/tecdsa/dkg accepting general tECDSA parameters.

Regarding the "silent dependency" I mean the following: if we have things like `PreParamsPoolSize` right in `pkg/tecdsa` that means this package is aware that there is something like a pre params pool in the DKG implementation. That information leaks from `pkg/tecdsa/dkg` and although it doesn't look as something dangerous now, it can lead to something weird in the future. If this was about something really general for tECDSA like, for example `threshold` or so, everything would be fine. I think the `pkg/tecdsa` package should remain a high-level one declaring only general concepts that are common for DKG and signing.

> I can put this struct in tbtc but then dkg.NewExecutor needs to accept params directly (not a struct).

I think this is fine. The `pkg/tbtc` is the application and it should be responsible for its own config (i.e. should declare it). Since it uses `pkg/tecdsa/dkg` directly, the knowledge about pre-params pool in `pkg/tbtc` seems to be justified.

Reply:
https://github.com/keep-network/keep-core/pull/3160/commits/6a805056561d1c1c85c62a3c8a3d92da62a63498 sir.